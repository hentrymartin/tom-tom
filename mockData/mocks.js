export const incident = {
  id: 'europe_HD_DE_TTL6000',
  type: 1,
  point: {
    x: 13.072789,
    y: 52.388527,
  },
  from: 'Am Buchhorst (Arthur-Scheunert-Allee/L78)',
  to: 'Potsdam-Brauhausberg - Friedrich-Engels-Stra√üe (Heinrich-Mann-Allee/L78)',
  details: 'stationary traffic',
  delay: 232,
  magnitude: 2,
};

export const incidents = [];
