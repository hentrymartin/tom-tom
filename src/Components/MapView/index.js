import React from 'react';
import {Images} from '../../Images';
import './MapView.scss';
import Incident from '../Incident';

const MapView = props => {
  return (
    <div className="map-view-wrapper">
      <img src={Images.MAP_IMAGE} alt="fake map" />
      <div className="incidents-board">
        {props.roadIncidents.map((incident, index) => (
          <Incident key={incident.id} incident={incident} />
        ))}
      </div>
    </div>
  );
};

MapView.defaultProps = {
  roadIncidents: [],
};

export default MapView;
