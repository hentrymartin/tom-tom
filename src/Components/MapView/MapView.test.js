import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import MapView from './index';
import { incidents } from '../../../mockData/mocks';

describe('Map View', () => {
  it('should render correctly', () => {
    const component = shallow(<MapView incidents={incidents} />);

    expect(component).toMatchSnapshot();
  });

  it('Check for number of incidents in the list', () => {
    const component = mount(
      <MemoryRouter>
        <MapView incidents={incidents} />
      </MemoryRouter>,
    );

    expect(component.find('.incident-wrapper').length).toBe(incidents.length);
  });
});
