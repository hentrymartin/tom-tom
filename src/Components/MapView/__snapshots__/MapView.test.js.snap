// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map View should render correctly 1`] = `ShallowWrapper {}`;
